src/main/java/com/snowplowanalytics/snowplow/tracker/emitter/BatchEmitter.java

```diff
@@ -91,6 +93,20 @@ public T bufferCapacity(final int bufferCapacity) {
             return self();
         }
 
+        /**
+         * Provide a denylist of HTTP response codes. Retry will not be attempted if one of these codes
+         * is received. The events in the request will be dropped. An exponential backoff time will
+         * be applied to subsequent requests. We advise, therefore, also setting a maximum bufferCapacity
+         * when providing fatal response codes, to avoid using too much memory storing unsendable events.
+         *
+         * @param fatalResponseCodes Event sending will not be retried on these codes
+         * @return itself
+         */
+        public T fatalResponseCodes(final List<Integer> fatalResponseCodes) {
```

Comment:
Ah interesting, you let the customer decide which response code should be fatal. My initial idea was to have a fixed list of fatal codes and the user can choose the behaviour of the tracker. Can't say what would be the best for the user. The responseCodes to add can be many.